### PR TITLE
[TextField: Spinner] Support press and hold to increment/decrement value

### LINKED ISSUE #420: Spinner on numeric TextField doesn't support press-and-hold
# Feature request summary
Make spinner buttons on numeric text fields respond to being pressed and held down, instead of only firing a single click event when the button is released.

## Rationale
Currently pressing and holding the spinner on a `TextField` with `type="number"` doesn't continually increment/decrement the value displayed:

numeric-text-field-spinner

However, we feel that most users would expect to be able to press and hold the spinner buttons to change the value more quickly than having to click multiple times. For example, in one of our apps, we have numeric text fields which control the positioning of a component on screen, and in that case, being able to hold the spinner down to "slide" the component around the screen is a useful ability to have.

### PR DESCRIPTION
**Pull request by tanhengyeow**
### WHY are these changes introduced?
Resolves #420

### WHAT is this pull request doing?

Support press and hold to increment/decrement value in `Spinner` of the numeric `TextField`.

## 

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import * as React from 'react';
import {Page, TextField} from '@shopify/polaris';

interface State {}

export default class Playground extends React.Component<never, State> {
  state = {
    value: '1',
  };

  handleChange = (value) => {
    this.setState({value});
  };

  render() {
    return (
      <Page title="Playground">
        {/* Add the code you want to test here */}
        <TextField
          label="Quantity"
          type="number"
          value={this.state.value}
          onChange={this.handleChange}
        />
      </Page>
    );
  }
}
```

</details>

### CODE REVIEW

File: src/components/TextField/TextField.tsx [hunk 1]
```diff
@@ -383,6 +388,16 @@ export default class TextField extends React.PureComponent<Props, State> {
   private handleClick() {
     this.input.focus();
   }
+
+  @autobind
+  private handleButtonPress(onChange: Function) {
+    this.buttonPressTimer = setInterval(onChange, 200);
```
Comment (tmlayton) [inline src/components/TextField/TextField.tsx]: `200` might be a bit long and I think we should consider adding a `setTimeout` first. That way when clicking and holding, there is an initial delay of maybe `500` then the number quickly increments maybe at an interval of `100`. Thoughts?

Comment (sarahill) [reply src/components/TextField/TextField.tsx]: ☝️ I like this suggestion of gradually increasing the speed. Overall, this looks great!

**Comment by dpersing** [comment]
Hi @tanhengyeow! Are you able to provide any updates around tests? Thanks!

**Comment by tanhengyeow** [comment]
Hey everyone, thanks for the review :smile:

The suggestion sounds great and I'll look into adding tests :) I'm having exams now so I can only get back to this during December, hope that's fine :D

**Comment by danrosenthal** [comment]
> I'm having exams now so I can only get back to this during December, hope that's fine

Totally fine. I'll add `[WIP]` to the title to indicate that this pull request is a work in progress.

**Comment by danrosenthal** [comment]
I might be missing something here, but when I tophat, it still just seems to increment and decrement by one digit.

**Comment by tmlayton** [comment]
When you `git log` do you see the latest commits? Depending on how you checked this out an upstream might not be set.

**Pull request merged**